perun-core/src/main/java/cz/metacentrum/perun/core/api/ServicesManager.java

```diff
@@ -264,6 +264,20 @@
 	 */
 	void deleteService(PerunSession perunSession, Service service) throws ServiceNotExistsException, PrivilegeException, RelationExistsException, ServiceAlreadyRemovedException;
 
+	/**
+	 * Deletes the service.
+	 *
+	 * @param perunSession
+	 * @param service
+	 * @param forceFlag
+	 * @throws InternalErrorException
+	 * @throws ServiceNotExistsException
+	 * @throws PrivilegeException
+	 * @throws RelationExistsException
+	 * @throws ServiceAlreadyRemovedException if there are 0 rows affected by deleting from DB
+	 */
+	void deleteService(PerunSession perunSession, Service service, boolean forceFlag) throws ServiceNotExistsException, PrivilegeException, RelationExistsException, ServiceAlreadyRemovedException;
```

Comment:
You don't really need to add a copy of the existing method with a new parameter into the API here. You can just add a force flag into the old one and take care of the RPC. The old way will use "false" as default if the user will not send such parameter into the API. It will prevent duplicates of code.

Reply:
OK, I have removed the old method altogether, RPC was already taken care of.